### PR TITLE
Us020/21 change respondent account status

### PR DESCRIPTION
**Pull request by gemmai95**
US020 and US021 add functionality to deactivate and re-activate a respondent's account. This works with changes in backstage and response-operations-ui.

### CODE REVIEW

**Comment by gemmai95** [comment]
[US021: INT Re-Activate respondents account (L) - Needed for R17](https://trello.com/c/MdXKJfBS/57-us021-int-re-activate-respondents-account-l-needed-for-r17)

File: ras_party/views/account_view.py [hunk 1]
```diff
@@ -86,3 +86,14 @@ def respondent_add_survey():
 
     ras_party.controllers.account_controller.add_new_survey_for_respondent(payload)
     return make_response(jsonify('OK'), 200)
+
+
+@account_view.route('/respondents/edit-account-status', methods=['PUT'])
```
Comment (andrewmil) [inline ras_party/views/account_view.py]: I think `party_id` should be in the url here rather than being passed in the body of the request

**Review by andrewmil — CHANGES_REQUESTED** [review]
Looks good :+1:

**Review by andrewmil — APPROVED** [review]

**Comment by gemmai95** [comment]
[US021: INT Re-Activate respondents account (L) - Needed for R17](https://trello.com/c/MdXKJfBS/57-us021-int-re-activate-respondents-account-l-needed-for-r17)

File: API.md [hunk 2]
```diff
@@ -259,6 +259,19 @@ This page documents the Party service API endpoints. All endpoints return an `HT
 ```
 &mdash; This endpoint will enrol a respondent in a survey and associate with business if not already associated.
 
+
+* `PUT /respondents/edit-account-status`
+
+### Example JSON data for put
+```json
+{
+    "party_id": "438df969-7c9c-4cd4-a89b-ac88cf0bfdf3",
```
Comment (jcox-dev) [inline API.md]: API docs will need amending to show that we're now including the party ID in the URL and not the JSON.

File: ras_party/controllers/account_controller.py [hunk 3]
```diff
@@ -291,6 +291,15 @@ def request_password_change(payload, session):
     return {'response': "Ok"}
 
 
+@with_db_session
+def change_respondent_account_status(payload, party_id, session):
+
+    status = payload['status_change']
+
+    respondent = query_respondent_by_party_uuid(party_id, session)
```
Comment (jcox-dev) [inline ras_party/controllers/account_controller.py:300]: Currently if party does not exist this raises a 500* error. Do we need to cover this (and return 404)? (And should we have a test for it?)

*
```python
respondent.status = status

AttributeError: 'NoneType' object has no attribute 'status'
```

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Doc update and potentially handling of missing party.

File: ras_party/views/account_view.py [hunk 4]
```diff
@@ -81,3 +81,14 @@ def respondent_add_survey():
...
+
+
+@account_view.route('/respondents/edit-account-status/<party_id>', methods=['PUT'])
+def put_respondent_account_status(party_id):
+    payload = request.get_json() or {}
+    v = Validator(Exists('status_change'))
+    if not v.validate(payload):
+        raise RasError(v.errors, 400)
+
+    ras_party.controllers.account_controller.change_respondent_account_status(payload, party_id)
```
Comment (tej99) [inline ras_party/views/account_view.py]: account_controller.change_respondent_account_status(payload, party_id)

**Review by jcox-dev — APPROVED** [review]

File: ras_party/controllers/account_controller.py [hunk 5]
```diff
@@ -291,6 +291,17 @@ def request_password_change(payload, session):
...
+    if not respondent:
+        raise RasError("Unable to find respondent account", status=404)
+    respondent.status = status
+
```
Comment (tej99) [inline ras_party/controllers/account_controller.py:304]: Are you wanting to return the respondent back from this function? if not the codacy error may need addressing.

**Review by tej99 — CHANGES_REQUESTED** [review]

File: ras_party/controllers/account_controller.py (hunk 5 above)
Comment (gemmai95) [reply ras_party/controllers/account_controller.py:304]: In process of changing this.

**Review by tej99 — APPROVED** [review]

**Pull request merged**